components/omega/test/infra/MetadataTest.cpp

```diff
@@ -0,0 +1,295 @@
+//===-- Test driver for OMEGA Metadata -------------------------*- C++ -*-===/
+//
+/// \file
+/// \brief Test driver for OMEGA Metadata
+///
+/// This driver tests the Metadata capabilities for the OMEGA
+/// model
+//
+//===-----------------------------------------------------------------------===/
+
+#include <iostream>
+#include <limits>
+
+#include "DataTypes.h"
+#include "Logging.h"
+#include "MetaData.h"
+
+using namespace OMEGA;
+
+void printResult(bool result, bool expected, std::string msg_pass,
+                 std::string msg_fail) {
+
+   if (result == expected) {
+      std::cout << msg_pass << ": PASS" << std::endl;
+   } else {
+      std::cout << msg_fail << ": FAIL" << std::endl;
+   }
+}
+
```

Comment:
@grnydawn Just one more change - in Omega, we'll be defining the metadata in various init routines and then retrieving them later in other routines. Can you create a testMetaInit routine in this test code that creates the metadata and then retrieve them later in the various other testMeta routines?  I'd like to make sure the use of shared_ptr does indeed retain the values after the initial variables go out of scope. Thanks.

Reply:
@philipwjones yes, I have added testMetaInit() and it looks working well to me.